src/disk_io_thread.cpp

```diff
@@ -2084,22 +2086,27 @@ namespace libtorrent
 		DLOG("kick_hasher: %d - %d (piece: %d offset: %d)\n"
```

Comment:
we set ``hashing = 1`` right above here. as far as I can tell, we don't release the mutex between checking ``hashing`` and setting it, meaning that there really only should be one thread getting here at a time, no?